gp_predict.f95

```diff
@@ -4406,6 +4410,28 @@ subroutine gpSparse_printXML(this,xf,label,sparseX_base_filename,error)
 
    endsubroutine gpSparse_printXML
 
+   subroutine gp_write_covariance(this, basename, label)
+      type(gpSparse), intent(in) :: this
+      character(*), intent(in) :: basename
+      character(*),intent(in), optional :: label
+
+
+      character(STRING_LENGTH) :: my_label, R_fname
+      integer :: M
+
+      if (present(label)) my_label = "." // trim(label)
```

Comment:
we have the `optional_default` interface for this. In this current form, `my_label` won't be initialised if `label` is not present.